shared/hoverAlerts.ts

```diff
@@ -0,0 +1,84 @@
+import * as sourcegraph from 'sourcegraph'
+
+export const HoverAlerts = {
+    LSIF: <sourcegraph.Badged<sourcegraph.HoverAlert>[]>[
```

Comment:
This casts the expression to the type, which means it silences any errors of missing properties. If you want to keep the `HoverAlerts` namespace, I would instead suggest to just do
```ts
export const lsif: sourcegraph.Badged<sourcegraph.HoverAlert>[] = [
  // ...
]
```
and import with
```ts
import * as HoverAlerts from './hoverAlerts'
```
(the difference is casting vs a type annotation)

There's an ESLint rule that shows this as a warning if you have some ESLint integration installed for your editor.

Reply:
ahh ok, will change.

for fun: before i added the casts, i was trying to create an object with every key of type Badged<HoverAlert>[] so I didn't have to write it every time. but Record<string, Badged<HoverAlert>[]> didn't work. do you know if there's a way to do this pattern in TS? basically a Map<K, V> with compile time enforced finite Ks?

Reply:
also, my style preference is to have constants start with upper case. my impression from our TS source is that types are uppercase, everything else is lowercase. is that a distinction we care about?